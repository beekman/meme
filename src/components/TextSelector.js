import React from 'react';
import PropTypes from 'prop-types';

const TextSelector = ({ headerText, image, footerText, onChange }) => (
  <>
    <input type="text" name="headerText" value={ headerText } onChange={ onChange } />
    <input type="text" name="image" value={ image } onChange={ onChange } />
    <input type="text" name="footerText" value={ footerText } onChange={ onChange } />
  </>
);

TextSelector.propTypes = {
  headerText: PropTypes.string.isRequired,
  footerText: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default TextSelector;
