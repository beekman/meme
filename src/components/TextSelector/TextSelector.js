import React from 'react';
import PropTypes from 'prop-types';
import styles from '../App.css';

const TextSelector = ({ headerText, image, footerText, onChange }) => (
  <div className={ styles.TextSelector }>
    <label>Image: <input type="text" name="image" value={ image } onChange={ onChange } /></label>
    <label>Header Text: <input type="text" name="headerText" value={ headerText } onChange={ onChange } /></label>
    <label>Footer Text: <input type="text" name="footerText" value={ footerText } onChange={ onChange } /></label>
  </div>
);

TextSelector.propTypes = {
  headerText: PropTypes.string.isRequired,
  footerText: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  onChange: PropTypes.func
};

export default TextSelector;
