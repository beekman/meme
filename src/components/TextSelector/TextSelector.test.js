import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from './TextSelector';

describe('TextSelector component', () => {
  it('renders TextSelector', () => {
    const wrapper = shallow(<TextSelector headerText="Hello" image="https://imgflip.com/s/meme/Futurama-Fry.jpg" footerText="Goodbye" />);
    expect(wrapper).toMatchSnapshot();
  });
});
