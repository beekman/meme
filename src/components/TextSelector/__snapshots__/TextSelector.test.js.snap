// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSelector component renders TextSelector 1`] = `
<div
  className="TextSelector"
>
  <label>
    Image: 
    <input
      name="image"
      type="text"
      value="https://imgflip.com/s/meme/Futurama-Fry.jpg"
    />
  </label>
  <label>
    Header Text: 
    <input
      name="headerText"
      type="text"
      value="Hello"
    />
  </label>
  <label>
    Footer Text: 
    <input
      name="footerText"
      type="text"
      value="Goodbye"
    />
  </label>
</div>
`;
